lib/customer-deposit-wallet/customer-deposit-wallet.cabal

```diff
@@ -60,6 +60,7 @@ library
     , cardano-strict-containers
     , cardano-wallet
     , cardano-wallet-network-layer
+    , cardano-wallet-primitive
```

Comment:
Can you avoid this pls.  Do we need the Interpreter or just some functions in `m` ?

Reply:
Not happy about it either, but I want the `TimeInterpreter` — it's reasonably well-designed. 🤔  One of the next steps would be to split off the `TimeInterpreter` into a separate package and get rid of the dependency on `-primitive` again.

Reply:
From the perspective of `createPayment` or a deposit wallet specific balanceTx wrapper I suspect you might ideally want to hide both `PParams` and `TimeTranslation` inside some single `LocalNodeTipStateForTxWrite`. Both for the sake of hiding the details, but also for the sake of querying it atomically from the node. But the read side will need the `TimeInterpreter` too.

Reply:
> also for the sake of querying it atomically from the node

For the deposit wallet this wouldn't matter though. In general, perhaps we still deem it not to matter, but it does lead to code like https://github.com/cardano-foundation/cardano-wallet/blob/aa7fe8c417bd133857f5eef44496cc5e603ed051/lib/wallet/src/Cardano/Wallet.hs#L2226-L2242

Reply:
I have no problem with the `m` as long as it's the way to get a pure functions out of it that we can reuse during atomic operations like ingesting a block or translating time in the tx history. 
But I would prefer to enumerate the exact functions we need out of it in the interface and leave the implementations of those to later. That would make mocking easier and avoid implementation leaks

Reply:
As discussed in a all, I have created a new indirection module `Cardano.Wallet.Deposit.Time` as a receptacle for slot-time-related stuff which we want to sort out later.

> but also for the sake of querying it atomically from the node

That's a fair point. Probably not in this pull request, though.